lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,536 @@
+
+        :param entity: Entity which we want to remove.
+        """
+        pass
+
+    def remove(self, entity=None, search_params=None, **kwargs):
+        """
+        Method which is called when state of the entity is 'absent'. If user
+        don't provide `entity` parameter the entity is searched using
+        `search_params` parameter. If entity is found it's removed.
+
+        Additional task can be executed after entity is searched and before
+        remove action is executed by overriding `pre_remove` method.
+
+        :param entity: Entity we want to remove.
+        :param search_params: Dictionary of parameters to be used for search.
+        :param kwargs: Additional parameters passed when removing entity.
+        :return: Dictionary with values returned by Ansible module.
+        """
+        if entity is None:
+            entity = self.search_entity(search_params)
+
+        if entity is None:
+            return {
+                'changed': self.changed,
+                'msg': "Entity wasn't found."
+            }
+
+        self.pre_remove(entity)
+
+        entity_service = self._service.service(entity.id)
+        if not self._module.check_mode:
+            entity_service.remove(**kwargs)
+
+            wait(
+                service=entity_service,
+                condition=lambda entity: not entity,
+                wait=self._module.params['wait'],
+                timeout=self._module.params['timeout'],
+            )
+        self.changed = True
+
+        return {
+            'changed': self.changed,
+            'id': entity.id,
+            type(entity).__name__.lower(): get_dict_of_struct(entity),
+        }
+
+    def action(
+        self,
+        action,
+        entity=None,
+        action_condition=lambda e: e,
+        wait_condition=lambda e: e,
+        pre_action=lambda e: e,
+        post_action=lambda e: None,
+        search_params=None,
+        **kwargs
+    ):
+        """
+        This method is executed when we want to change the state of some oVirt
+        entity. The action to be executed on oVirt service is specified by
+        `action` parameter. Whether the action should be executed can be
+        specified by passing `action_condition` parameter. Which state the entity
+        should have after executing the action can be specified by passing
+        `wait_condition` parameter.
+
+        Additional task can be executed before action on entity is executed
+        by passing `pre_action` parameter to method. Another task can be
+        executed after action is executed, by passing `post_action` parameter
+        to method.
+
+        :param action: Action which should be executed on service.
+        :param entity: Entity we want to run action on.
+        :param action_condition: Function which is executed when checking if action should be executed.
+        :param wait_condition: Function which is executed when waiting on result state.
```

Comment:
That's something I didn't think about, it's great idea, the current flow wouldn't work well enough if it would be part of the _wait_condition_. I will definitelly add it, thanks.